File: packages/common/erp-common-ms-utils-js/src/helpers/jwt.ts
import jwt from 'jsonwebtoken';
import { IAuth } from '@gmahechas/erp-common-js';
import { env } from './env';
import { sendError } from '../errors/utils/send-error';
import { TypeErrorMessage } from '../errors/utils/error-type.enum';
import { envKeysType } from '../interfaces';

export const jwtSign = (
  payload: string | object | SourceBufferEventMap,
  secretOrPrivateKey: jwt.Secret,
  options?: jwt.SignOptions | undefined,
) => jwt.sign(payload, secretOrPrivateKey, options);

export const jwtVerify = (
  token: string,
  secretOrPublicKey: jwt.Secret,
  options?: jwt.SignOptions | undefined,
) => jwt.verify(token, secretOrPublicKey, options);

export const jwtDecode = (token: string) => {
  const appName = process.env.APP_NAME as envKeysType;
  if (!appName) {
    sendError(TypeErrorMessage.CONFIG);
  }
  const publicKey = env[appName]!.auth!.jwt!.publicKey!;
  if (!publicKey) {
    sendError(TypeErrorMessage.CONFIG);
  }
  try {
    return jwt.verify(token, publicKey) as IAuth;
  } catch (error) {
    sendError(TypeErrorMessage.AUTHENTICATION);
  }
};

File: packages/common/erp-common-ms-utils-py/gmahechas_erp_common_ms_utils_py/helpers/jwt.py
import os
import jwt
from .env import Env


class JWT:

    @staticmethod
    def verify(token, secret_or_public_key):
        return jwt.decode(token, secret_or_public_key, algorithms="RS256")

    @staticmethod
    def decode(token):
        app_name = os.environ["APP_NAME"]
        public_key = Env.get()[app_name]["auth"]["jwt"]["publicKey"]
        return JWT.verify(token, public_key)
